indico/modules/rb/client/js/common/rooms/RoomEditDetails.jsx

```diff
@@ -0,0 +1,48 @@
+import {Form, Header} from 'semantic-ui-react';
+import React from 'react';
+import {FavoritesProvider} from 'indico/react/hooks';
+import {FinalPrincipal} from 'indico/react/components';
+import {Translate} from 'indico/react/i18n';
+import {FinalInput, FinalTextArea} from 'indico/react/forms';
+
+function RoomEditDetails() {
+  return (
+    <>
+      <Header>
+        <Translate>Contact</Translate>
+      </Header>
+      <FavoritesProvider>
+        {favoriteUsersController => (
```

Comment:
Not needed; you can use the `useFavoriteUsers` hook directly here; the only reason `FavoritesProvider` exists is to use it in class-based components.

Reply:
Also, unless this component stays mounted even when switching tabs I would move this hook one level up into the parent component and pass the controller as a prop - otherwise the favorite users ids are resolved every time this component gets mounted.

Reply:
> Also, unless this component stays mounted even when switching tabs I would move this hook one level up into the parent component and pass the controller as a prop - otherwise the favorite users ids are resolved every time this component gets mounted.

Unfortunately, the component needs to stay mounted when switching tabs to maintain the form state consistent. Nonetheless, I think I may end up moving it up just to have more control and keep this unidirectional flow.